code/__DEFINES/projectiles.dm

```diff
@@ -73,3 +73,9 @@
 
 #define RETURN_POINT_VECTOR(ATOM, ANGLE, SPEED) (new /datum/point/vector(ATOM, null, null, null, null, ANGLE, SPEED))
 #define RETURN_POINT_VECTOR_INCREMENT(ATOM, ANGLE, SPEED, AMT) (new /datum/point/vector(ATOM, null, null, null, null, ANGLE, SPEED, AMT))
+
+/// The amount of energy that a standard energy weapon cell can hold
+#define ENERGY_WEAPON_MAX_CHARGE 1000
+/// Macro to turn a number of laser shots into an energy cost, based on the above define
+/// e.g. LASER_SHOTS(12) means 12 shots
+#define LASER_SHOTS(X) (((100 * ENERGY_WEAPON_MAX_CHARGE) - ((100 * ENERGY_WEAPON_MAX_CHARGE) % X)) / (100 * X)) // I wish I could just use round, but it can't be used in datum members
```

Comment:
That isn't a foolproof approach, since it assumes the energy weapon has a standard power cell installed, which is not always the case. If anything, it needs a second argument.
```suggestion
#define LASER_SHOTS(X, MAX_CHARGE) (((100 * MAX_CHARGE) - ((100 * MAX_CHARGE) % X)) / (100 * X)) // I wish I could just use round, but it can't be used in datum members
```